### DISCUSSION TITLE
Updating only one dependency

### DISCUSSION THREAD

**Discussion by calvinbui** [renovatebot/renovate#31067] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

We use renovate to distribute updates for internal/private packages. When we trigger renovate, it would also update other dependencies (go, terraform, python, etc). which are listed in the repo's `enabledManagers`. Is there a way to override the renovate config?

I have tried:

1. `requireConfig: ignored`: marks existing PRs as `abandoned`. 
1. setting `RENOVATE_ENABLED_MANAGERS` on renovate: I believe the repo's config overrides it

Possible solutions:

1. Get renovate to ignore the repo's config and use a custom preset/config (still causes the abandoned error)

**Comment by RahulGautamSingh** [comment]
Can't you just remove the other managers from `enabledManagers` ?


If not then try this:

```json
{
"packageRules": [
   {
     "matchPackageNames": ["*"],
     "enabled": false
   },
   {
     "matchPackageNames": ["name-of-dep-you-want-to-update"],
     "enabled": true
   }
 ]
}
```

If you want more support, please share your config. Or, update the description to be more precise.

**Reply by calvinbui** [reply]
is it possible to override the config file when renovate is running?

**Reply by RahulGautamSingh** [reply]
Repo config?

**Reply by calvinbui** [reply]
yes the repository's config

**Reply by RahulGautamSingh** [reply]
https://docs.renovatebot.com/config-overview/#force-config

**Reply by calvinbui** [reply]
thanks. how would i workaround renovate marking all existing PRs are abandoned? is that based on a setting (branchPrefix maybe?)

**Reply by RahulGautamSingh** [reply]
No, the bot abandons any pr which needs no update & has been modified. Which could be true for your case.

It seems you run renovate twice on some repos. Once you want to update only one dep and on another run you are updating the rest.
But since they run on the same repo, they effect previous run's changes.

Is this correct? If not, please explain your case. Its hard to help without context.

**Reply by calvinbui** [reply]
ideal workflow:

1. renovate does a normal run over the weekend and generates PRs as configured in each repo's `renovate.json5`. This creates PRs for go modules, terraform versions, helm charts, etc.
2. I merge all those PRs and deploy them on Monday (patch day). Some PRs I do not merge, as they require more testing or have breaking changes, etc.
3. During the week, I update a private npm module and wish to get that deployed (i.e. v1.0.0 -> v2.0.0).
4. I trigger renovate again, which creates the PR to update the npm module, but also creates PRs for dependencies that have updated since the last run over the weekend.
5. I don't want renovate to also create PRs for go modules, terraform versions, etc. again, because patching day is Monday.
6. If I configure renovate to update only the private npm module, it marks the earlier PRs I did not merge as abandoned.

**Reply by RahulGautamSingh** [reply]
Try using `schedule`. On the run that you update the private dep ... configure the schedule for all the other to somewhere unreachable. This way they won't get updated during that run.

**Reply by rarkins** [reply]
Maybe also needs updateNotScheduled=false too

**Reply by RahulGautamSingh** [reply]
Yeah, to supress all/any updates this option neeeds to be set as well

**Reply by calvinbui** [reply]
thanks for all the help. this is my working solution:

renovate env vars:

```
RENOVATE_REPOSITORIES = "calvinbui/update-this"
RENOVATE_INHERIT_CONFIG = true
RENOVATE_INHERIT_CONFIG_REPO_NAME = "calvinbui/renovate-configs"
RENOVATE_INHERIT_CONFIG_FILE_NAME = "custom-docker-images.json5"
```

my config at `calvinbui/renovate-configs` named `custom-docker-images.json5`:

```json5
// ignore the repo's configuration file
requireConfig: "ignored",

// dont update any existing PRs
updateNotScheduled: false,
pruneStaleBranches: false,

<all my package rules and customManagers>
```
